### ISSUE TITLE
Error in mrs_imatch sigma-clipping calls

### ISSUE THREAD

**Issue by stscijgbot-jp** [spacetelescope/jwst#4938]
_Issue [JP-1450](https://jira.stsci.edu/browse/JP-1450) was created on JIRA by [Howard Bushouse](https://jira.stsci.edu/secure/ViewProfile.jspa?name=bushouse):_

B7.5 testing has encountered a few errors in calwebb_spec3 processing of MIRI MRS exposures in the mrs_imatch step. The traceback is:
```java
2020-05-11 23:09:22,240 - stpipe.Spec3Pipeline.mrs_imatch - INFO - Update S_REGION to POLYGON ICRS  98.827264159 -66.834495332 98.827264159 -66.832653665 98.831027756 -66.832653665 98.831027756 -66.834495332
Traceback (most recent call last):
  File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-0.16.0.20200507-py37/bin/strun", line 26, in <module>
    step = Step.from_cmdline(sys.argv[1:])
  File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-0.16.0.20200507-py37/lib/python3.7/site-packages/jwst/stpipe/step.py", line 156, in from_cmdline
    return cmdline.step_from_cmdline(args)
  File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-0.16.0.20200507-py37/lib/python3.7/site-packages/jwst/stpipe/cmdline.py", line 346, in step_from_cmdline
    step.run(*positional)
  File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-0.16.0.20200507-py37/lib/python3.7/site-packages/jwst/stpipe/step.py", line 392, in run
    step_result = self.process(*args)
  File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-0.16.0.20200507-py37/lib/python3.7/site-packages/jwst/pipeline/calwebb_spec3.py", line 177, in process
    result = self.mrs_imatch(result)
  File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-0.16.0.20200507-py37/lib/python3.7/site-packages/jwst/stpipe/step.py", line 392, in run
    step_result = self.process(*args)
  File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-0.16.0.20200507-py37/lib/python3.7/site-packages/jwst/mrs_imatch/mrs_imatch_step.py", line 82, in process
    _match_models(single_ch[c], channel=str(c), degree=degree)
  File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-0.16.0.20200507-py37/lib/python3.7/site-packages/jwst/mrs_imatch/mrs_imatch_step.py", line 342, in _match_models
    _, themed, clipsig = sigclip(thisimg, mask_value=0.)
  File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-0.16.0.20200507-py37/lib/python3.7/site-packages/astropy/stats/sigma_clipping.py", line 756, in sigma_clipped_stats
    mean = np.nanmean(data_clipped, axis=axis)
  File "<__array_function__ internals>", line 6, in nanmean
  File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-0.16.0.20200507-py37/lib/python3.7/site-packages/numpy/lib/nanfunctions.py", line 952, in nanmean
    avg = _divide_by_count(tot, cnt, out=out)
  File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-0.16.0.20200507-py37/lib/python3.7/site-packages/numpy/lib/nanfunctions.py", line 214, in _divide_by_count
    return np.divide(a, b, out=a, casting='unsafe')
ValueError: output array is read-only
----------------------------------------------------------------------
ERROR RUNNING STEP 'Spec3Pipeline':
    output array is read-only
----------------------------------------------------------------------
```
which appears to be a problem encountered down in the numpy functions being used to do sigma clipping. Seems like we've encountered this before and thought we had a workaround in place. Needs investigating again. Keep in mind it's highly to be due to garbage data.

**Comment by stscijgbot-jp** [comment]
_Comment by [James Davies](https://jira.stsci.edu/secure/ViewProfile.jspa?name=jdavies) on [JIRA](https://jira.stsci.edu/browse/JP-1450?focusedCommentId=463808#comment-463808):_

This looks like the same issue that was seen here

<https://jira.stsci.edu/browse/JP-1231> ([github](https://github.com/spacetelescope/jwst/issues/4451))

and reported in `astropy`

astropy/astropy#9973

And was fixed in `astropy`

astropy/astropy#10099

which will go into astropy 4.0.2.  Not released yet.  So we should probably avoid sending fully masked or NaN arrays to `sigma_clipped_stats` much as we did  in JP-1231 above.

**Comment by stscijgbot-jp** [comment]
_Comment by [Howard Bushouse](https://jira.stsci.edu/secure/ViewProfile.jspa?name=bushouse) on [JIRA](https://jira.stsci.edu/browse/JP-1450?focusedCommentId=463811#comment-463811):_

Same fix that was applied to outlier_detection in <https://github.com/spacetelescope/jwst/pull/4598/files> could be applied here in mrs_imatch.

**Comment by stscijgbot-jp** [comment]
_Comment by [James Davies](https://jira.stsci.edu/secure/ViewProfile.jspa?name=jdavies) on [JIRA](https://jira.stsci.edu/browse/JP-1450?focusedCommentId=463812#comment-463812):_

This fix in `outlier_detection` was

<https://github.com/spacetelescope/jwst/pull/4598/files#diff-50083546cda96a27c1b44bc5b8ee20a6R201-R208>

**Comment by stscijgbot-jp** [comment]
_Comment by [Howard Bushouse](https://jira.stsci.edu/secure/ViewProfile.jspa?name=bushouse) on [JIRA](https://jira.stsci.edu/browse/JP-1450?focusedCommentId=463813#comment-463813):_

Does anyone know an ETA for astropy 4.0.2?

**Comment by stscijgbot-jp** [comment]
_Comment by [James Davies](https://jira.stsci.edu/secure/ViewProfile.jspa?name=jdavies) on [JIRA](https://jira.stsci.edu/browse/JP-1450?focusedCommentId=463880#comment-463880):_

Sometime very soon, but not in time for Build 7.5 delivery of `jwst` I suspect.

**Comment by stscijgbot-jp** [comment]
_Comment by [Howard Bushouse](https://jira.stsci.edu/secure/ViewProfile.jspa?name=bushouse) on [JIRA](https://jira.stsci.edu/browse/JP-1450?focusedCommentId=463957#comment-463957):_

Fixed by #4944
